src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -843,39 +859,31 @@ public override InterfaceMapping GetInterfaceMap([DynamicallyAccessedMembers(Dyn
             ValidateInterfaceType(interfaceType);
 
             MethodInfo[] interfaceMethods = interfaceType.GetMethods(BindingFlags.DeclaredOnly | BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static);
-            InterfaceMapping im = default;
-            im.InterfaceType = interfaceType;
-            im.TargetType = this;
-            im.InterfaceMethods = new MethodInfo[interfaceMethods.Length];
-            im.TargetMethods = new MethodInfo[interfaceMethods.Length];
+            InterfaceMapping im = new InterfaceMapping
+            {
+                InterfaceType = interfaceType,
+                TargetType = this,
+                InterfaceMethods = new MethodInfo[interfaceMethods.Length],
+                TargetMethods = new MethodInfo[interfaceMethods.Length]
+            };
 
             for (int i = 0; i < interfaceMethods.Length; i++)
             {
                 MethodInfo interfaceMethod = interfaceMethods[i];
                 MethodInfo? targetMethod = GetMethodImpl(interfaceMethod.Name, GetBindingFlags(interfaceMethod), null, interfaceMethod.CallingConvention, GetParameterTypes(interfaceMethod.GetParameters()), null);
 
-                if (targetMethod == null)
-                {
-                    Type? parent = _typeParent;
```

Comment:
Because this logic moved to `GetMethodImpl` that called above at row 873 https://github.com/dotnet/runtime/pull/95753/files/a1fd35b35409c74e262f5adb123ff0d30cbf2fdc..7fbc17b3c3e3789baa298225d2f81c5a8fa05c9e#diff-c12b00ae2e7f16430a305235a8e8ea053806a0d3fc4b15ea4e18f5c0cb904cfaR811-R821
i.e. when `BindingFlags.DeclaredOnly` is not set `GetMethodImpl` will look within parent methods too